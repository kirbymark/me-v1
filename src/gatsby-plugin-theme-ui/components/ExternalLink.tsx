/** @jsx jsx */
import * as React from "react"
import { jsx, ThemeUICSSObject } from "theme-ui"
import theme from "../index"

const externalLinkStyles: ThemeUICSSObject = {
  color: `primary`,
  textDecorationLine: `none`,
  "&:hover": { textDecorationLine: `underline` },
}

const ExternalLink = ({children, ...props}) => (
  <a sx={{ ...externalLinkStyles }} target="_blank" rel="noopener noreferrer" {...props}>
    {children}
  </a>
)

export default ExternalLink

