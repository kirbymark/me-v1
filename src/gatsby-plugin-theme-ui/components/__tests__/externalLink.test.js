import React from "react"
import renderer from "react-test-renderer"

import ExternalLink from "../externalLink"

describe("ExternalLink", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ExternalLink href="https://linuxhandbook.com/homelab/">Homelab</ExternalLink>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})