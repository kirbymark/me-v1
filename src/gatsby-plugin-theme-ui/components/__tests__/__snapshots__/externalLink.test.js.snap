// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders correctly 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-n2y63d-ExternalLink{color:primary;text-decoration-line:none;}.css-n2y63d-ExternalLink:hover{text-decoration-line:underline;}",
      }
    }
    data-emotion="css n2y63d-ExternalLink"
  />,
  <a
    className="css-n2y63d-ExternalLink"
    href="https://linuxhandbook.com/homelab/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Homelab
  </a>,
]
`;
